### PR TITLE
QIIME2 workflows

### PR DESCRIPTION
**Pull request by debjyoti197**
- [ ] Input
  - [x] Input demultiplexed data single and paired end
  - [x] Input multiplexed data single and paired end
  - [ ] Other imports https://forum.qiime2.org/t/importing-and-demultiplexing-sequence-data-quick-reference/14002
- [x] Denoising
- [ ] Phylogeny
- [ ] Diversity Analysis
- [ ] Taxonomic Classification

### CODE REVIEW

**Comment by bernt-matthias** [comment]
Let me add my own TODO list :)

- [ ] for the input workflows the sample data should be limited `fastq.gz` .. or better `fastqsanger.gz`  (since the tool doesn't do this yet)
- [ ] the Vizualisations should be workflow outputs (I guess), other wise we can not test them.

**Comment by bernt-matthias** [comment]
Hi @mvdbeek. Besides the `inputs` in the workflows this should be ready for review. All other (downstream) workflows will be in https://github.com/galaxyproject/iwc/pull/372

File: workflows/amplicon/qiime2/qiime2-I-import/.dockstore.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  subclass: Galaxy
+  publish: true
+  primaryDescriptorPath: /QIIME2-Ia-multiplexed-data-single-end.ga
+  testParameterFiles:
+  - /QIIME2-Ia-multiplexed-data-single-end-tests.yml
+  authors:
+  - name: Debjyoti Ghosh
+    orcid: 0009-0008-1496-1677
+  - name: "Helmholtz-Zentrum f\xFCr Umweltforschung - UFZ"
+    address: "Permoserstra\xDFe 15, 04318 Leipzig"
```
Comment (mvdbeek) [inline workflows/amplicon/qiime2/qiime2-I-import/.dockstore.yml:13]: That's interesting, can you fix the umlaut here ? Seems like a bug we should fix.

File: workflows/amplicon/qiime2/qiime2-I-import/QIIME2-Ia-multiplexed-data-single-end.ga [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+            "name": "Debjyoti Ghosh"
+        },
+        {
+            "address": "Permoserstra\u00dfe 15, 04318 Leipzig",
+            "class": "Organization",
+            "name": "Helmholtz-Zentrum f\u00fcr Umweltforschung - UFZ"
+        }
+    ],
+    "format-version": "0.1",
+    "license": "MIT",
```
Comment (mvdbeek) [inline workflows/amplicon/qiime2/qiime2-I-import/QIIME2-Ia-multiplexed-data-single-end.ga:17]:
```suggestion
    "license": "MIT",
    "release": "0.1",
```

File: workflows/amplicon/qiime2/qiime2-I-import/QIIME2-Ic-demultiplexed-data-single-end.ga [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+            "name": "Debjyoti Ghosh"
+        },
+        {
+            "address": "Permoserstra\u00dfe 15, 04318 Leipzig",
+            "class": "Organization",
+            "name": "Helmholtz-Zentrum f\u00fcr Umweltforschung - UFZ"
+        }
+    ],
+    "format-version": "0.1",
+    "license": "MIT",
```
Comment (mvdbeek) [inline workflows/amplicon/qiime2/qiime2-I-import/QIIME2-Ic-demultiplexed-data-single-end.ga:17]:
```suggestion
    "license": "MIT",
    "release": "0.1",
```

File: workflows/amplicon/qiime2/qiime2-II-denoising/QIIME2-IIa-denoising-and-feature-table-creation-single-end.ga [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+            "identifier": "0009-0008-1496-1677",
+            "name": "Debjyoti Ghosh"
+        },
+        {
+            "address": "Permoserstra\u00dfe 15, 04318 Leipzig",
+            "class": "Organization",
+            "name": "Helmholtz-Zentrum f\u00fcr Umweltforschung - UFZ"
+        }
+    ],
+    "format-version": "0.1",
```
Comment (mvdbeek) [inline workflows/amplicon/qiime2/qiime2-II-denoising/QIIME2-IIa-denoising-and-feature-table-creation-single-end.ga:17]:
```suggestion
    "format-version": "0.1",
    "release": "0.1",
```

File: workflows/amplicon/qiime2/qiime2-II-denoising/QIIME2-IIb-denoising-and-feature-table-creation-paired-end.ga [hunk 5]
```diff
@@ -0,0 +1,438 @@
...
+            "identifier": "0009-0008-1496-1677",
+            "name": "Debjyoti Ghosh"
+        },
+        {
+            "address": "Permoserstra\u00dfe 15, 04318 Leipzig",
+            "class": "Organization",
+            "name": "Helmholtz-Zentrum f\u00fcr Umweltforschung - UFZ"
+        }
+    ],
+    "format-version": "0.1",
```
Comment (mvdbeek) [inline workflows/amplicon/qiime2/qiime2-II-denoising/QIIME2-IIb-denoising-and-feature-table-creation-paired-end.ga:17]:
```suggestion
    "format-version": "0.1",
    "release": "0.1",
```

File: workflows/amplicon/qiime2/qiime2-II-denoising/README.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# QIIME2 workflows
+
+## Available workflows
+
+Denoising (using `qiime2`'s `dada2` integration for paired / single end data.
+
+## Inputs
+
+- Demultiplexed sequences a a qiime2 aertifact file (`qza`) containing the sequence information.
```
Comment (mvdbeek) [inline workflows/amplicon/qiime2/qiime2-II-denoising/README.md:9]:
```suggestion
- Demultiplexed sequences as a qiime2 aertifact file (`qza`) containing the sequence information.
```

File: workflows/amplicon/qiime2/qiime2-I-import/QIIME2-Ib-multiplexed-data-paired-end.ga [hunk 7]
```diff
@@ -15,6 +15,7 @@
     ],
     "format-version": "0.1",
     "license": "MIT",
+    "release" "0.1",
```
Comment (mvdbeek) [inline workflows/amplicon/qiime2/qiime2-I-import/QIIME2-Ib-multiplexed-data-paired-end.ga:18]: Oops, missing colon here :(

File: workflows/amplicon/qiime2/qiime2-I-import/QIIME2-Ib-multiplexed-data-paired-end.ga [hunk 8]
```diff
@@ -0,0 +1,433 @@
...
+        },
+        {
+            "address": "Permoserstra\u00dfe 15, 04318 Leipzig",
+            "class": "Organization",
+            "name": "Helmholtz-Zentrum f\u00fcr Umweltforschung - UFZ"
+        }
+    ],
+    "format-version": "0.1",
+    "license": "MIT",
+    "release" "0.1",
```
Comment (mvdbeek) [inline workflows/amplicon/qiime2/qiime2-I-import/QIIME2-Ib-multiplexed-data-paired-end.ga:18]:
```suggestion
    "release": "0.1",
```

File: workflows/amplicon/qiime2/README.md [hunk 9]
```diff
@@ -0,0 +1,10 @@
+# QIIME2 workflows
+
+## Available workflows
+
+1. Import of fastqsanger.gz data into QIIME artifact files. Available for
+   - paired / single end data
+   - demultiplexed / multiplexed data (the later according to the EMP protocol)
```
Comment (mvdbeek) [inline workflows/amplicon/qiime2/README.md:7]:
```suggestion
   - demultiplexed / multiplexed data (the latter according to the EMP protocol)
```

**Review by mvdbeek — APPROVED** [review]
Thanks a lot, looks great!

**Pull request merged**